overlord/servicestate/helpers.go

```diff
@@ -21,25 +21,80 @@ package servicestate
 
 import (
 	"fmt"
+	"os/user"
 	"sort"
+	"strconv"
 
+	"github.com/snapcore/snapd/client/clientutil"
 	"github.com/snapcore/snapd/overlord/snapstate"
 	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/wrappers"
 )
 
-// updateSnapstateServices uses ServicesEnabledByHooks and ServicesDisabledByHooks in
-// snapstate and the provided enabled or disabled list to update the state of services in snapstate.
-// It is meant for doServiceControl to help track enabling and disabling of services.
-func updateSnapstateServices(snapst *snapstate.SnapState, enable, disable []*snap.AppInfo) (bool, error) {
-	if len(enable) > 0 && len(disable) > 0 {
-		// We do one op at a time for given service-control task; we could in
-		// theory support both at the same time here, but service-control
-		// ops are run sequentially so we always either enable or disable at
-		// any given time. Not having to worry about that simplifies the
-		// problem of ordering of enable vs disable.
-		return false, fmt.Errorf("internal error: cannot handle enabled and disabled services at the same time")
+var userLookup = user.Lookup
+
+// usernamesToUids converts a list of usernames, to a list of uids by
+// looking up the usernames.
+func usernamesToUids(usernames []string) ([]int, error) {
```

Comment:
we have similar code in wrappers/services.go, would it make sense to move it to osutil?

Reply:
moved this